Library/Homebrew/resource.rb

```diff
@@ -266,15 +266,15 @@ def stage_resource(prefix, debug_symbols: false, &block)
   private
 
   def detect_version(val)
-    return Version::NULL if val.nil? && url.nil?
-
-    case val
-    when nil     then Version.detect(url, **specs)
+    version = case val
+    when nil     then url.nil? ? Version::NULL : Version.detect(url, **specs)
```

Comment:
```suggestion
    when nil     then url.present? ? Version.detect(url, **specs) : Version::NULL
```

although, given this `version.null?` is getting discarded below: could it just be:

```suggestion
    when nil     then Version.detect(url, **specs) if url.present?
```

Reply:
`Version.detect` can return NULL itself.

Reply:
Ok. The first feels a bit nicer to me if you agree?